src/Common/src/Interop/Windows/mincore/Interop.GetVersionExW.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.SystemInfo_L1_1, CharSet = CharSet.Unicode, SetLastError = true)]
+        internal static extern bool GetVersionExW(ref OSVERSIONINFOEX osvi);
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        internal struct OSVERSIONINFOEX
+        {
+            public int dwOSVersionInfoSize;
+            public int dwMajorVersion;
+            public int dwMinorVersion;
+            public int dwBuildNumber;
+            public int dwPlatformId;
+            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 128)]
```

Comment:
`public fixed char szCSDVersion[128];` would be better - faster, and you can also use `sizeof(OSVERSIONINFOEX)` instead of the slow `Marshal.SizeOf` to initialize the size.